Quadriga/src/main/java/edu/asu/spring/quadriga/aspects/ConceptCollectionAuthorization.java

```diff
@@ -30,28 +33,25 @@
     @Autowired
     private IConceptCollectionFactory collectionFactory;
 
-    /**
-     * This method checks the access permissions for the given ConceptCollection
-     * for the logged in user.
-     * 
-     * @param userName
-     *            - logged in user name.
-     * @param conceptCollectionId
-     *            - Concept Collection id.
-     * @param userRoles
-     *            - the roles for which the user should be checked against the
-     *            Concept Collection.
-     * @throws QuadrigaAccessException
-     *             ,QuadrigaStorageException
-     * @author Kiran Batna
-     */
+    private final Logger logger = LoggerFactory.getLogger(ConceptCollectionAuthorization.class);
+
     @Override
-    public boolean chkAuthorization(String userName,
-            String conceptCollectionId, String[] userRoles)
+    public boolean chkAuthorization(String userName, Object conceptCollectionObj, String[] userRoles)
             throws QuadrigaStorageException, QuadrigaAccessException {
-        
+
+        IConceptCollection collection;
+        String conceptCollectionId = null;
         // fetch the details of the concept collection
-        IConceptCollection collection = conceptCollectionManager.getConceptCollection(conceptCollectionId);
+        try {
+            if (conceptCollectionObj.getClass().equals(String.class)) {
+                conceptCollectionId = (String) conceptCollectionObj;
+                collection = conceptCollectionManager.getConceptCollection(conceptCollectionId);
+            } else {
+                collection = (IConceptCollection) conceptCollectionObj;
+            }
+        } catch (ClassCastException cce) {
```

Comment:
Well, only the if statement will be out of the block, do you want me to put two different try...catch blocks?